src/napari_matplotlib/line.py

```diff
@@ -0,0 +1,246 @@
+from typing import List, Optional, Tuple
+
+import matplotlib.colors as mcolor
+import napari
+import numpy as np
+import pandas as pd
+from magicgui import magicgui
+from magicgui.widgets import ComboBox
+
+from .base import NapariMPLWidget
+from .util import Interval
+
+__all__ = ["Line2DBaseWidget", "MetadataLine2DWidget"]
+
+
+class Line2DBaseWidget(NapariMPLWidget):
+    # opacity value for the lines
+    _line_alpha = 0.5
+    _lines = []
+
+    def __init__(self, napari_viewer: napari.viewer.Viewer):
+        super().__init__(napari_viewer)
+
+        self.axes = self.canvas.figure.subplots()
+        self.update_layers(None)
+
+    def clear(self) -> None:
+        """
+        Clear the axes.
+        """
+        self.axes.clear()
+
+    def draw(self) -> None:
+        """
+        Plot the currently selected layers.
+        """
+        data, x_axis_name, y_axis_name = self._get_data()
+        
+
+        if len(data) == 0:
+            # don't plot if there isn't data
+            return
+        self._lines = []
+        x_data = data[0]
+        y_data = data[1]
+
+        if len(y_data) < len(x_data):
+            print("x_data bigger than y_data, plotting only first y_data")
+        for i, y in enumerate(y_data):
+            if len(x_data) == 1:
+                line = self.axes.plot(x_data[0], y, alpha=self._line_alpha)
+            else:
+                line = self.axes.plot(x_data[i], y, alpha=self._line_alpha)
+            self._lines += line
+
+        self.axes.set_xlabel(x_axis_name)
+        self.axes.set_ylabel(y_axis_name)
+
+    def _get_data(self) -> Tuple[List[np.ndarray], str, str]:
+        """Get the plot data.
+
+        This must be implemented on the subclass.
+
+        Returns
+        -------
+        data : np.ndarray
+            The list containing the line plot data.
+        x_axis_name : str
+            The label to display on the x axis
+        y_axis_name: str
+            The label to display on the y axis
+        """
+        raise NotImplementedError
+
+class MetadataLine2DWidget(Line2DBaseWidget):
+    n_layers_input = Interval(1, 1)
+
+    def __init__(self, napari_viewer: napari.viewer.Viewer):
+        super().__init__(napari_viewer)
+        self.setMinimumSize(200, 200)
+        self._plugin_name_widget = magicgui(
```

Comment:
I'm confused about what the `plugin_name` stuff is for - could you explain?

Reply:
This was because I thought about a scenario where there are multiple plugins saving/loading settings/features in a layer metadata. If these features have the same name, there would be overwriting problems. That is also why I started with a nested dictionary, as a way to isolate them. What do you think?
We could also drop this level, it would be more up to the user to know that this can happen.